src/index.ts

```diff
@@ -7,120 +7,134 @@ metadata = {
                 }
             }
         }
-)};
+    }
+    )
+};
 
-onexecute = function(objectName, methodName, parameters, properties) {
-    switch (objectName)
-    {
+onexecute = async function (objectName, methodName, parameters, properties): Promise<void> {
+    switch (objectName) {
         case "message": onexecuteMessage(methodName, parameters, properties); break;
         default: throw new Error("The object " + objectName + " is not supported.");
     }
 }
 
- function onexecuteMessage(methodName: string, parameters: SingleRecord, properties: SingleRecord) {
-    switch (methodName)
-    {
+async function onexecuteMessage(methodName: string, parameters: SingleRecord, properties: SingleRecord): Promise<void> {
+    switch (methodName) {
         case "get": onexecuteMessageGet(parameters, properties); break;
         case "list": onexecuteMessageList(parameters, properties); break;
         default: throw new Error("The method " + methodName + " is not supported.");
     }
 }
 
-function onexecuteMessageGet(parameters: SingleRecord, properties: SingleRecord) {
-    var xhr = new XMLHttpRequest();
+function onexecuteMessageGet(parameters: SingleRecord, properties: SingleRecord): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+        var xhr = new XMLHttpRequest();
 
-    xhr.onreadystatechange = function() {
-            if (xhr.readyState !== 4) return;
-            if (xhr.status !== 200) throw new Error("Failed with status " + xhr.status);
+        xhr.onreadystatechange = function () {
+            try {
+                if (xhr.readyState !== 4) return;
+                if (xhr.status !== 200) throw new Error("Failed with status " + xhr.status);
 
-            //console.log(xhr.responseText);
-            var obj = JSON.parse(xhr.responseText);
-            postResult({
-                "id": obj.id,
-                "subject": obj.subject,
-                "from": obj.from.emailAddress.address,
-                "body": obj.body.content
-            });            
-    };
+                //console.log(xhr.responseText);
+                var obj = JSON.parse(xhr.responseText);
+                postResult({
+                    "id": obj.id,
+                    "subject": obj.subject,
+                    "from": obj.from.emailAddress.address,
+                    "body": obj.body.content
+                });
+                resolve();
+            } catch (error) {
+                reject(error);
+            }
+        };
 
-    var url = "https://graph.microsoft.com/v1.0/users/" + encodeURIComponent(parameters["userPrincipalName"]) + "/mailfolders%28%27Inbox%27%29/messages/" + encodeURIComponent(parameters["id"]);
-    //console.log(url);
-    xhr.open("GET", url);
+        var url = "https://graph.microsoft.com/v1.0/users/" + encodeURIComponent(parameters["userPrincipalName"]) + "/mailfolders%28%27Inbox%27%29/messages/" + encodeURIComponent(parameters["id"]);
+        //console.log(url);

```

Comment:
Should this be removed?

Reply:
are you talking about the commented lines. Cleanup was not a part of the ticket

Reply:
but I can remove them

Reply:
done